docs/services/gcp-services-guide.md

```diff
@@ -0,0 +1,536 @@
+```
+
+### Crossplane
+Using your GKE cluster:
+
+* Install Crossplane from alpha channel using the [Crossplane Installation
+  Guide](../install-crossplane.md#master)
+* Install the GCP stack into Crossplane using the [GCP stack
+  section](../install-crossplane.md#gcp-stack) of the install guide.
+
+We will be using `gcp-infra-dev` namespace to store our cloud-specific resource
+classes, cloud resources and providers. But you can use any namespace name for
+that matter.
+```bash
+kubectl create namespace gcp-infra-dev
+export INFRA_NAMESPACE=gcp-infra-dev
+```
+
+#### GCP Provider
+It is essential to make sure that a GCP service account to be used by 
+Crossplane is configured with all the necessary permissions outlined in the 
+[provider guide](../cloud-providers/gcp/gcp-provider.md).
+
+Using the service account json `crossplane-gcp-provider-key.json`:
+* Generate Base64 encoded value to store in a `Secret`:
+```bash
+export BASE64ENCODED_GCP_PROVIDER_CREDS=$(base64 crossplane-gcp-provider-key.json | tr -d "\n")
+```
+
+* Define a GCP `Provider` and `Secret`:
+```bash
+cat > gcp-provider.yaml <<EOF
+---
+apiVersion: v1
+data:
+  credentials.json: $BASE64ENCODED_GCP_PROVIDER_CREDS
+kind: Secret
+metadata:
+  namespace: $INFRA_NAMESPACE
+  name: gcp-provider-creds
+type: Opaque
+---
+apiVersion: gcp.crossplane.io/v1alpha2
+kind: Provider
+metadata:
+  namespace: $INFRA_NAMESPACE
+  name: gcp-provider
+spec:
+  credentialsSecretRef:
+    name: gcp-provider-creds
+    key: credentials.json
+  projectID: $PROJECT_ID
+EOF
+
+kubectl apply -f gcp-provider.yaml
+
+```
+* Verify GCP provider was successfully registered by the crossplane
+```bash
+kubectl get providers.gcp.crossplane.io -n $INFRA_NAMESPACE
+kubectl get secrets -n $INFRA_NAMESPACE
+```
+
+#### Cloud-Specific Resource Classes
+Cloud-specific resource classes are used to define a reusable configuration for
+a specific managed service. Wordpress requires a MySQL database, which can be
+satisfied by a [Google Cloud SQL Instance](https://cloud.google.com/sql/docs/mysql/) instance.
+
+* Define a GCP CloudSQL class `CloudsqlInstanceClass`:
+```bash
+cat > gcp-mysql-standard.yaml <<EOF
+---
+apiVersion: database.gcp.crossplane.io/v1alpha2
+kind: CloudsqlInstanceClass
+metadata:
+  name: standard-cloudsql
+  namespace: $INFRA_NAMESPACE
+specTemplate:
+  databaseVersion: MYSQL_5_7
+  tier: db-n1-standard-1
+  region: us-central1
+  storageType: PD_SSD
+  storageGB: 10
+  # Note from GCP Docs: Your Cloud SQL instances are not created in your VPC network.
+  # They are created in the service producer network (a VPC network internal to Google) that is then connected (peered) to your VPC network.
+  privateNetwork: projects/$PROJECT_ID/global/networks/$NETWORK_NAME
+  providerRef:
+    name: gcp-provider
+    namespace: $INFRA_NAMESPACE
+  reclaimPolicy: Delete
+EOF
+
+kubectl apply -f gcp-mysql-standard.yaml
+```
+* You can verify creation with the following command and output:
+
+*Command*
```

Comment:
The pattern I usually like to use is:

```
$ kubectl get cloudsqlinstanceclass -n ${INFRA_NAMESPACE}
NAME                   PROVIDER-REF   RECLAIM-POLICY   AGE
standard-cloudsql      gcp-provider   Delete           11s
```

Because that's what it looks like when people are in the terminal. Sometimes I'll put `> ` at the beginning of the stuff that's printed out if there's only one line of output, or if there's another prompt.

Reply:
I actually don't like that pattern because it doesn't allow you to just select the whole line and copy&paste into the terminal, you'd have to be careful to select only the command. I think k8s docs are also prepared this way https://kubernetes.io/docs/tutorials/stateless-application/guestbook/#creating-the-redis-master-deployment

Reply:
I agree with your point. Being able to copy and paste more easily is the way to go; we should separate input and output everywhere!

Reply:
+1 on this, I think it's a much better experience when you can copy the entire block to paste and run it in your terminal, and not have to be careful about which parts you're selecting.